scan/lib/scan/test_command_generator.rb

```diff
@@ -49,7 +50,9 @@ def actions
         actions = []
         actions << :clean if config[:clean]
         actions << :build unless config[:skip_build]
-        actions << :test
+        actions << "build-for-testing" if config[:build_for_testing]
+        actions << :test unless config[:test_without_building] || config[:build_for_testing]
```

Comment:
@lyricsboy you are right this looks like a good thing. in fact we still need some more detection of this, as with build-for-testing/test-without-building we need to switch parameters here: https://github.com/fastlane/fastlane/pull/7130/files#diff-ef49192ad6b55e081d8efc1ddabd1ee8R36

i ll happily adopt to proposed solution - but i think there is still no core-team decision on how to proceed with the whole , `build-for-testing` thing. @KrauseFx  any idea?

@mfurtak probably 8.2 brings a speed improvement - should we retest it?